### PR TITLE
Creates a "Writing a spike" document for the conda organization

### PR DESCRIPTION
**Pull request by travishathaway**
This pull request contains a document that is intended to be used by everyone working on issues within the conda organization (or all projects under https://github.com/conda).

I welcome discussion and edits to ensure this document is agreed to by everyone currently working on conda projects.

### CODE REVIEW

File: ISSUE_SPIKE.md [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+Another common way that we create spikes is during the planning and discovery phase of an epic.
+Epics represent larger chunks of work that need to be spread out over multiple issues. Spikes are
+often needed to figure out the exact details for how a particular feature is to be implemented
+because this may not be known when we first begin work on the epic.
+
+### How do I create a spike?
+
+#### When converting an existing issue
+
+- Preface the issue title with, "SPIKE 🔍" (or add a label???)
```
Comment (kathatherine) [inline ISSUE_SPIKE.md:39]:
```suggestion
- Preface the issue title with "SPIKE 🔍" (or add a label???)
```

File: ISSUE_SPIKE.md [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+- Ensure there is enough background information on the issue for a would-be assignee
+- Ensure that the issue has clear deliverables. These should state exactly what needs to be done
+  For example, if we expect to create another issue as result of this spike, that should be stated
+
+#### When planning an epic
+
+- Preface the issue title with, "SPIKE 🔍" (or add a label???)
```
Comment (kathatherine) [inline ISSUE_SPIKE.md:46]:
```suggestion
- Preface the issue title with "SPIKE 🔍" (or add a label???)
```

**Review by kathatherine — CHANGES_REQUESTED** [review]
Those commas aren't necessary. Great explanation!

File: ISSUE_SPIKE.md [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+because this may not be known when we first begin work on the epic.
+
+### How do I create a spike?
+
+#### When converting an existing issue
+
+- Preface the issue title with, "SPIKE 🔍" (or add a label???)
+- Ensure there is enough background information on the issue for a would-be assignee
+- Ensure that the issue has clear deliverables. These should state exactly what needs to be done
+  For example, if we expect to create another issue as result of this spike, that should be stated
```
Comment (beeankha) [inline ISSUE_SPIKE.md:42]:
```suggestion
- Ensure that the issue has clear deliverables; these should state exactly what needs to be done (e.g., if we expect to create another issue as result of this spike, that should be stated)
```

File: ISSUE_SPIKE.md [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+  knowledge into deliverables for other issues
+- Ensure that the deliverables have been clearly defined and whether other issues
+  need to be created as a result of this spike
+
+### How do I complete a spike?
+
+Determining whether a spike is complete should be an easy task provided
+that the deliverables have been clearly defined. When this is not the case, you should
+not work on the issue and instead attempt to clarify the deliverables with those also
+working on the project.
```
Comment (beeankha) [inline ISSUE_SPIKE.md:58]:
```suggestion
Determining whether a spike is complete should be a straightforward task provided
that the deliverables have been clearly defined. When this is not the case, you should
not work on the issue and instead attempt to clarify the deliverables with those who are
also working on the project.
```

**Review by beeankha — CHANGES_REQUESTED** [review]
We need to decide on whether or not we'll have official "spike" labels, so let's figure out the `(or add a label???)` bits before we merge this PR

File: SPIKES.md [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+### When do I **not** create a spike?
+
+Below are some common scenarios where creating a spike is not appropriate:
+
+- Writing a technical specification
+- Design work that would go into drafting how a library our API is going to be created
+- Any work that must be completed or is not optional
+
+### How do I create a spike?
+
```
Comment (kenodegard) [inline SPIKES.md:65]:
```suggestion

Spikes start as a task associated with a regular issue. Only create a spike issue if we determine the spike to be a larger endeavor with multiple deliverables, see the guidelines below.

```

**Review by kenodegard — CHANGES_REQUESTED** [review]
Looks good, some suggestions on what we call the different kinds of "issues"

File: SPIKES.md (hunk 5 above)
Comment (kathatherine) [reply SPIKES.md:65]: Who's "we" here?

Also "see the guidelines below" should be its own sentence.

File: SPIKES.md [hunk 6]
```diff
@@ -68,11 +76,6 @@ instead be created as regular issues.
...
-
-### How do I complete a spike?
-
-Determining whether a spike is complete should be an easy task provided
-that the deliverables have been clearly defined. When this is not the case, you should
-not work on the issue and instead attempt to clarify the deliverables with those also
-working on the project.
+  related issue.
+- Spikes need deliverables. If there are no deliverables listed, do not work on the spike and instead
+  determine what these deliverables are before going forward and update the spike.
```
Comment (kathatherine) [inline SPIKES.md:81]:
```suggestion
- Spikes need deliverables. If there are no deliverables listed, do not work on the spike and instead
  determine what these deliverables are before going forward and updating the spike.
```

File: SPIKES.md [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+
+```mermaid
+flowchart LR
+
+A[Create Spike] --> B[Work on spike for\n timebox amount, \n update if necessary]
+B --> C{is it done?}
+C -->|No| E{do we want\nto continue?}
+C -->|Yes| D[Finished]
+E -->|Yes | B
+E -->|No| D
```
[the remark this answers is not in the document]
Comment (travishathaway) [reply SPIKES.md:42]: Thank @kenodegard, this is my first time creating a chart with mermaid. I will reference that style chart next time.

File: SPIKES.md (hunk 6 above)
Comment (travishathaway) [reply SPIKES.md:81]: @kathatherine, here I'm going to stick with what I wrote because I mean it like:

Do not work on the spike and instead:
- determine ....
- update the spike

Hope that makes sense 😄

File: SPIKES.md (hunk 5 above)
Comment (travishathaway) [reply SPIKES.md:73]: @kenodegard, I have at least one issue with this: spikes don't always start as tasks for a regular issue. For example, creating a prototype wouldn't necessarily always start off with an associated issue. So, let's make sure to accommodate for this as well. 

Perhaps it may be better to first describe what a "mini" spike looks like and then what a spike looks like. Here's how a "mini" spike could look like on a regular issue:

- [ ] Spike description (time box: _1 day_)
  - [ ] Create a report with your finding from the spike; include recommendation if relevant

☝️ As you can see it includes the time box and what we expect as the deliverable. We can assume issue itself would have enough background information.

File: SPIKES.md [hunk 8]
```diff
@@ -0,0 +1,89 @@
...
+come in which deal with problems or technologies that our development team is unfamiliar with.
+For all issues which our development team has sufficient knowledge of, these issues should
+instead proceed as regular issues.
+
+### When do I **not** create a spike?
+
+Below are some common scenarios where creating a spike is not appropriate:
+
+- Writing a technical specification
+- Design work that would go into drafting how a library our API is going to be created
```
Comment (beeankha) [inline SPIKES.md:69]:
```suggestion
- Design work that would go into drafting how a library in our API is going to be created
```
(actually I'm not sure if this is the correct edit, but as it is currently this sentence needs to be clarified a bit)

File: SPIKES.md [hunk 9]
```diff
@@ -0,0 +1,89 @@
...
+- Any work that must be completed or is not optional
+
+### How do I create a spike?
+
+#### All spikes need to following information
```
Comment (beeankha) [inline SPIKES.md:74]:
```suggestion
#### Requirements for all spikes
```

File: SPIKES.md [hunk 10]
```diff
@@ -0,0 +1,89 @@
...
+
+- The "spike" label
+- A timebox between 0 and 3 days (e.g. 4 hours; 2 days)
+- Background information for a would-be assignee
+- Clear deliverables. These should state exactly what needs to be done.
+  For example, if we expect to create another issue as result of this spike, that should be stated.
```
Comment (beeankha) [inline SPIKES.md:80]:
```suggestion
  For example, if we expect to create another issue as a result of this spike, that should be stated.
```

File: SPIKES.md (hunk 6 above)
Comment (beeankha) [reply SPIKES.md:89]: With your clarification in mind, I suggest rewording this sentence like:

```
Spikes need deliverables. If there are no deliverables listed, do not work on the spike; instead, determine what these deliverables are and then update the spike with that information.
```

File: SPIKES.md [hunk 11]
```diff
@@ -0,0 +1,89 @@
...
+to either research the problem or create a prototype in order to better understand it.
+
+Additionally, spikes represent work that **may or may not actually be completed
+or implemented**. An example of this are prototypes created to explore possible solutions.
+Not all prototypes are implemented and the purpose of creating a prototype
+is often to explore the problem space more. For research oriented tasks, the end result
+of this research may be that a feature request simply is not viable at the moment and
+would result in putting a stop to that work.
+
+Finally, spikes are **time boxed**. This means that we set a definite limit on how long
```
Comment (beeankha) [inline SPIKES.md:24]:
```suggestion
Finally, spikes are **timeboxed**. This means that we set a definite limit on how long
```
"Timebox" is written as one word in the rest of this doc so I'm suggesting this change to keep things consistent

File: SPIKES.md (hunk 6 above)
Comment (kathatherine) [reply SPIKES.md:89]: I agree with Bianca's edit.

File: SPIKES.md (hunk 11 above)
Comment (travishathaway) [reply SPIKES.md:24]: That word doesn't appear in my dictionary, so I'm going to switch to using the two word version.

File: SPIKES.md (hunk 8 above)
Comment (travishathaway) [reply SPIKES.md:69]: Thanks for pointing this out. I think I will go with:
```suggestion
- Design work that would go into drafting how a part of our API is going to be created
```

File: SPIKES.md (hunk 6 above)
Comment (travishathaway) [reply SPIKES.md:89]: Thanks @beeankha!

```suggestion
- Spikes need deliverables. If there are no deliverables listed, do not work on the spike; instead, determine what these deliverables are and then update the spike with that information.
```

File: HOW_WE_USE_GITHUB.md [hunk 12]
```diff
@@ -238,17 +238,180 @@ please post details to the [Nucleus forums](https://community.anaconda.cloud/).
...
+to perform further work on the spike.
+
+Below is what that workflow looks like:
+
+```mermaid
+flowchart LR
+
+%% nodes
+create[Create spike]
+work[Work on spike for\ntime box amount]
```
Comment (jezdez) [inline HOW_WE_USE_GITHUB.md:292]:
```suggestion
work[Timeboxed work on spike]
```

File: HOW_WE_USE_GITHUB.md [hunk 13]
```diff
@@ -238,17 +238,180 @@ please post details to the [Nucleus forums](https://community.anaconda.cloud/).
...
+to either research the problem or create a prototype in order to better understand it.
+
+Additionally, spikes represent work that **may or may not actually be completed
+or implemented**. An example of this are prototypes created to explore possible solutions.
+Not all prototypes are implemented and the purpose of creating a prototype
+is often to explore the problem space more. For research oriented tasks, the end result
+of this research may be that a feature request simply is not viable at the moment and
+would result in putting a stop to that work.
+
+Finally, spikes are **time boxed**. This means that we set a definite limit on how long
```
Comment (jezdez) [inline HOW_WE_USE_GITHUB.md:277]: The agile technique is called timeboxing, so I think it's correct to say `timeboxed`, right?

```suggestion
Finally, spikes are **timeboxed**. This means that we set a definite limit on how long
```

File: HOW_WE_USE_GITHUB.md [hunk 14]
```diff
@@ -238,17 +238,180 @@ please post details to the [Nucleus forums](https://community.anaconda.cloud/).
...
+we want our contributors to work on a spike. We do this to prevent contributors
+from falling into a rabbit hole they may never return from (scary!). Instead, we set
+a time limit, typically between 1 and 3 "days", to perform work on the spike and then
+have the assignee report back to the development team. If the tasks defined in
+the spike have not yet been completed, a decision is made on whether it makes sense
+to perform further work on the spike.
```
Comment (jezdez) [inline HOW_WE_USE_GITHUB.md:283]: Given the asynchronous nature of Open Source work, "days" are a bad scale for any contribution formats, including spikes. It suffices to say work on spikes need to be timeboxed and decided case by case to be inclusive to volunteers and non-staff contributors.

```suggestion
from falling into a rabbit hole they may never return from (scary!). Instead, we set
a time limit to perform work on the spike and then have the assignee report back to
the development team. If the tasks defined in the spike have not yet been completed,
a decision is made on whether it makes sense to perform further work on the spike.
```

File: HOW_WE_USE_GITHUB.md [hunk 15]
```diff
@@ -238,17 +238,180 @@ please post details to the [Nucleus forums](https://community.anaconda.cloud/).
...
+done -->|Yes| finished
+continue -->|Yes,\nset new time box| work
+continue -->|No| finished
+```
+
+### When do I create a spike?
+
+A spike should be created when we do not have enough information to move forward with solving
+a problem. That simply means whenever we are dealing with unknowns or processes which our
+development team has never encountered before, it may be useful for us to create a spike.
```
Comment (jezdez) [inline HOW_WE_USE_GITHUB.md:310]:
```suggestion
a problem. That simply means whenever we are dealing with unknowns or processes which the
development team has never encountered before, it may be useful for us to create a spike.
```

File: HOW_WE_USE_GITHUB.md [hunk 16]
```diff
@@ -238,17 +238,180 @@ please post details to the [Nucleus forums](https://community.anaconda.cloud/).
...
+
+In day-to-day work, this may appear when new bug reports or feature requests
+come in which deal with problems or technologies that our development team is unfamiliar with.
+For all issues which our development team has sufficient knowledge of, these issues should
+instead proceed as regular issues.
```
Comment (jezdez) [inline HOW_WE_USE_GITHUB.md:315]:
```suggestion
In day-to-day work, this may appear when new bug reports or feature requests
come in which deal with problems or technologies that the development team is unfamiliar with.
For all issues which the development team has sufficient knowledge of, these issues should
instead proceed as regular issues.
```

File: HOW_WE_USE_GITHUB.md [hunk 17]
```diff
@@ -238,17 +238,180 @@ please post details to the [Nucleus forums](https://community.anaconda.cloud/).
...
+done{Is it done?}
+finished[Finished]
+continue{More time?}
+
+%% graph
+create --> work
+work --> done
+done -->|No| continue
+done -->|Yes| finished
+continue -->|Yes,\nset new time box| work
```
Comment (jezdez) [inline HOW_WE_USE_GITHUB.md:302]:
```suggestion
continue -->|Yes,\nrestart timebox| work
```

File: HOW_WE_USE_GITHUB.md [hunk 18]
```diff
@@ -238,17 +238,180 @@ please post details to the [Nucleus forums](https://community.anaconda.cloud/).
...
+
+### How do I create and complete a spike?
+
+The following section describes the exact process of creating and completing
+a spike. There are two types of spikes to be aware of: "task spikes" and "issue spikes".
+We cover these two types of spikes as well as some examples below.
+
+#### Requirements for all spikes
+
+- A time box between one and three days (e.g. four hours; two days)
```
Comment (jezdez) [inline HOW_WE_USE_GITHUB.md:333]:
```suggestion
- A timebox, at most one sprint long.
```

File: HOW_WE_USE_GITHUB.md [hunk 19]
```diff
@@ -238,17 +238,180 @@ please post details to the [Nucleus forums](https://community.anaconda.cloud/).
...
+- Background information for a would-be assignee
+
+```
+Quick note about time boxes:
+
+In "time box" time, one day is the amount of hours in a normal work day (eight hours).
+This does not need to be completed consecutively. So, it is possible to complete a
+spike which has been time boxed for one day over the course of three or four days.
```
Comment (jezdez) [inline HOW_WE_USE_GITHUB.md:341]:
```suggestion
Quick note about timeboxes:

As the length of a timebox we only set a maximum of one sprint given the asynchronous nature of Open Source projects like conda.
```

File: HOW_WE_USE_GITHUB.md [hunk 20]
```diff
@@ -238,17 +238,180 @@ please post details to the [Nucleus forums](https://community.anaconda.cloud/).
...
+
+---
+
+The above example has several tasks which may take considerable time to complete; therefore, it
+warrants its own separate issue. It is also important to note that because this is an issue for
+a prototype, the work is optional and can be discontinued at any time. Further work on it will
+be determined by feedback from the project team or conda community.
+
+Just like task spikes, issue spike time boxes can also be renegotiated and reset if the project
+team feels that further work is necessary or warranted.
```
Comment (jezdez) [inline HOW_WE_USE_GITHUB.md:409]: This is way too complicated for regular contributors and represents a serious barrier of entry, we should dumb this text down so it become easy to understand and is less bureaucratic.

**Review by jezdez — CHANGES_REQUESTED** [review]
This needs another pass since the way this is written, it'll add a serious barrier of entry for contributors, basically it's too bureaucratic to be useful for an Open Source project.

Could we dumb down "task spikes" and "issue spikes" to just "spike issues" and stop using "days" as a scale for contributions? I think it's not reasonable to do that in a globally distributed, asynchronous Open Source project. Who would measure this? Will drive-by contributors feel compelled to follow these patterns?

Remember the goal of Open Source projects is to make it easy for community members to follow these instructions and not have to jump through hoops to work on conda.

File: HOW_WE_USE_GITHUB.md (hunk 18 above)
Comment (beeankha) [reply HOW_WE_USE_GITHUB.md:333]: I think we should specify that we mean "two weeks" when we say "sprint" here, so maybe:

`- A timebox, at most one sprint long (i.e., two weeks).`

File: HOW_WE_USE_GITHUB.md (hunk 14 above)
Comment (kenodegard) [reply HOW_WE_USE_GITHUB.md:283]: I don't expect contributors to bother with spikes, I see it more as an informative document for anyone who is confused about what we are doing with our planning/what a spike means.

Perhaps we should remove all units from this doc like you're suggesting and include a note admonition that specifies the units we use on our team (e.g., 2-week sprints, "days" for our timeboxes, etc.)?

File: HOW_WE_USE_GITHUB.md (hunk 18 above)
Comment (kenodegard) [reply HOW_WE_USE_GITHUB.md:333]: See suggestion for to use an admonition for our team specific units in https://github.com/conda/infra/pull/611#discussion_r984800996

File: HOW_WE_USE_GITHUB.md (hunk 20 above)
Comment (travishathaway) [reply HOW_WE_USE_GITHUB.md:409]: @jezdez, I believe you are bringing up a good point.

This document is only intended for those us who spend a majority of their working hours on conda projects (everything under https://github.com/conda). Regular contributors (those submitting bug-fixes or suggesting enhancements) will not be pressured to follow these guidelines. Regardless, these individuals will still benefit by having this information available as it helps them better understand the core development team's project management practices.

Therefore, I would very much prefer not having this text simplified in anyway.

File: HOW_WE_USE_GITHUB.md (hunk 14 above)
Comment (jezdez) [reply HOW_WE_USE_GITHUB.md]: I'm trying to understand why spikes wouldn't be something we could encourage contributors (especially when they are paid to contribute)?

E.g. when they request a feature, they might be able to help with the request by spending time on researching the topic more than the initial issue. I think Jaime and Filipe have both done them, just like many other people in the past.

If we're documenting the Ursa team's choices for "agile development" here, I think this will be seen by potential contributors as the de-facto way to contribute, even if we're not requiring it.

I'd suggest removing the timeframes and just document spikes as "research issues" devoid of any more profound meaning.

File: HOW_WE_USE_GITHUB.md (hunk 20 above)
Comment (jezdez) [reply HOW_WE_USE_GITHUB.md:408]: I've mentioned this in an Anaconda-internal chat as well: by making the document structured in this way, we're setting the expected mode of contribution to be “high barrier” but we need to keep it at “low barrier” (or if possible lowering it) to include everyone in the community and scale the ecosystem.

Maybe if this were just about the conda repo, or just for conda-build, it'd be fine, since project teams are free to self-govern and find their best practices. But this document here defines a specific style of agile development for the conda community infrastructure, and that's written too specific for that.

It's also wrong to say that this document would only for people "who spend a majority of their working hours on conda projects (everything under https://github.com/conda)" since
a) if it becomes a file in the conda organization repos, of course it will be seen as a requirement for everyone
b) not everyone working on projects under the conda organization are working in their working hours
c) the people providing feedback here (so far) are not representing the people working on projects under the conda organization

In short I think it's repeating the same mistakes of the past of defining community processes by fiat and not proposals.

**Comment by travishathaway** [comment]
@jezdez, I went back and tried to re-word the "Creating tasks/issues" sections as best as I possibly could. This time, I’m trying to put the “contributor” at the center of it without thinking about whether or it is a "core" developer or someone brand new to the project who just wants to help out.

I know the process is a little more involved, but I actually think that defining timeboxes for would-be contributors could be a really good thing. Imagine you’d like to contribute but only have maybe two or four extra hours to spare. If a project team makes sure to apply the correct time boxes to spikes, then you would be able to pick the right one for you.

I also think that having clear deliverables defined on issues (not just spikes) is very helpful for everyone involved. New and old contributors alike. This way, everyone knows exactly what is expected of them and how to successfully complete an issue.

I also replace usage of “development team” with “project team” as I think that reflects reality a little better. I also made sure to avoid the usage of the term "sprint" as that might not apply to all project teams.

File: HOW_WE_USE_GITHUB.md [hunk 21]
```diff
@@ -238,17 +238,189 @@ please post details to the [Nucleus forums](https://community.anaconda.cloud/).
...
 TODO
 
-#### Spikes
+## Spikes
 
-TODO
+In this section, we go over the procedure to follow when creating spikes. Exactly what
+these types of issues are and the steps to take when creating them are all outlined
+below. These guidelines apply for all projects falling underneath the
+[conda](github.com/conda) organization.
```
Comment (beeankha) [inline HOW_WE_USE_GITHUB.md:261]:
```suggestion
[`conda`](github.com/conda) organization.
```

File: HOW_WE_USE_GITHUB.md [hunk 22]
```diff
@@ -238,17 +238,189 @@ please post details to the [Nucleus forums](https://community.anaconda.cloud/).
...
+
+### What is a spike?
+
+A spike is a term that is borrowed from extreme programming and agile development.
```
Comment (beeankha) [inline HOW_WE_USE_GITHUB.md:265]:
```suggestion
"Spike" is a term that is borrowed from extreme programming and agile development.
```

File: HOW_WE_USE_GITHUB.md [hunk 23]
```diff
@@ -238,17 +238,189 @@ please post details to the [Nucleus forums](https://community.anaconda.cloud/).
...
+They are used when the **outcome of a ticket is unknown or even optional**. For example,
+when first coming across a problem that has not been solved before, a project may choose
+to either research the problem or create a prototype in order to better understand it.
+
+Additionally, spikes represent work that **may or may not actually be completed
+or implemented**. An example of this are prototypes created to explore possible solutions.
+Not all prototypes are implemented and the purpose of creating a prototype
+is often to explore the problem space more. For research oriented tasks, the end result
```
Comment (beeankha) [inline HOW_WE_USE_GITHUB.md:273]:
```suggestion
is often to explore the problem space more. For research-oriented tasks, the end result
```

File: HOW_WE_USE_GITHUB.md [hunk 24]
```diff
@@ -238,17 +238,189 @@ please post details to the [Nucleus forums](https://community.anaconda.cloud/).
...
+
+Our timebox levels try to give project teams and would-be contributors an idea
+of how much work is desired/expected for a given spike.  Our three timebox levels
+are defined as follows:
+
+- **Low** two to six hours of work
+- **Medium** - six to 12 hours of work
+- **High** - 12 to 24 hours of work
+
+It is important to note that these hours do not have to be completely consecutively, but
```
Comment (beeankha) [inline HOW_WE_USE_GITHUB.md:345]:
```suggestion
It is important to note that these hours do not have to be completed consecutively, but
```

File: HOW_WE_USE_GITHUB.md [hunk 25]
```diff
@@ -238,17 +238,189 @@ please post details to the [Nucleus forums](https://community.anaconda.cloud/).
...
+The example above outlines all the steps necessary in order to create a prototype to present to
+the conda community or project team. For this particular spike, the authors have specified its
+timebox as "high", so as a contributor, you should plan to spend a considerable amount of time
+on this spike.
+
+As someone working on a spike such as this, you are also highly encourage to keep all applicable
+information and discussions on this issue. This helps encourage a culture of asnychronous communication,
+and also helps us keep public records of important conversations that can be easily referenced later.
+
+#### Additional guidelines to be aware of
```
Comment (beeankha) [inline HOW_WE_USE_GITHUB.md:420]: This header should either be:
`#### Additional guidelines to be aware of:`
or 
`#### Additional guidelines`

**Review by beeankha — APPROVED** [review]
Other than a few more minor suggestions, this LGTM!

File: HOW_WE_USE_GITHUB.md (hunk 21 above)
Comment (kathatherine) [reply HOW_WE_USE_GITHUB.md:261]: According to our style guide, backticks are only necessary when talking about the command, not the org, package manager, etc.

Comment (beeankha) [reply HOW_WE_USE_GITHUB.md:261]: Thanks for the clarification, @kathatherine !

File: HOW_WE_USE_GITHUB.md (hunk 25 above)
Comment (travishathaway) [inline HOW_WE_USE_GITHUB.md:420]:
```suggestion
#### Additional guidelines
```

File: HOW_WE_USE_GITHUB.md [hunk 26]
```diff
@@ -238,17 +238,189 @@ please post details to the [Nucleus forums](https://community.anaconda.cloud/).
...
+### When do I create a spike?
+
+A spike should be created when we do not have enough information to move forward with solving
+a problem. That simply means whenever we are dealing with unknowns or processes which the
+project team has never encountered before, it may be useful for us to create a spike.
+
+In day-to-day work, this may appear when new bug reports or feature requests
+come in which deal with problems or technologies that the project team is unfamiliar with.
+For all issues which the project team has sufficient knowledge of, these issues should
+instead proceed as regular issues.
```
Comment (kathatherine) [inline HOW_WE_USE_GITHUB.md:314]:
```suggestion
In day-to-day work, this kind of situation may appear when new bug reports or feature 
requests come in that deal with problems or technologies that the project team is 
unfamiliar with. All issues that the project team has sufficient knowledge of should 
instead proceed as regular issues.
```

File: HOW_WE_USE_GITHUB.md [hunk 27]
```diff
@@ -238,17 +238,189 @@ please post details to the [Nucleus forums](https://community.anaconda.cloud/).
...
+
+### How do I create and complete a spike?
+
+The following section describes the exact process of creating and completing
+a spike. There are two types of spikes to be aware of: "task spikes" and "issue spikes".
+We cover these two types of spikes as well as some examples below.
+
+#### Requirements for all spikes
+
+- A timebox at a level of either "low", "medium" or "high" (more on this below)
```
Comment (kathatherine) [inline HOW_WE_USE_GITHUB.md:332]:
```suggestion
- A timebox at a level of either "low", "medium", or "high" (more on this below)
```

File: HOW_WE_USE_GITHUB.md [hunk 28]
```diff
@@ -238,17 +238,189 @@ please post details to the [Nucleus forums](https://community.anaconda.cloud/).
...
+- Background information for a would-be assignee
+
+#### Timebox levels
+
+Our timebox levels try to give project teams and would-be contributors an idea
+of how much work is desired/expected for a given spike.  Our three timebox levels
```
Comment (kathatherine) [inline HOW_WE_USE_GITHUB.md:338]:
```suggestion
of how much work is desired/expected for a given spike. Our three timebox levels
```

File: HOW_WE_USE_GITHUB.md [hunk 29]
```diff
@@ -238,17 +238,189 @@ please post details to the [Nucleus forums](https://community.anaconda.cloud/).
...
+are defined as follows:
+
+- **Low** two to six hours of work
```
Comment (kathatherine) [inline HOW_WE_USE_GITHUB.md:341]:
```suggestion
- **Low** - two to six hours of work
```

File: HOW_WE_USE_GITHUB.md [hunk 30]
```diff
@@ -238,17 +238,189 @@ please post details to the [Nucleus forums](https://community.anaconda.cloud/).
...
+- **Medium** - six to 12 hours of work
+- **High** - 12 to 24 hours of work
+
+It is important to note that these hours do not have to be completed consecutively, but
+you are encouraged to keep the timelines of your project team in mind if applicable.
```
Comment (kathatherine) [inline HOW_WE_USE_GITHUB.md:346]:
```suggestion
you are encouraged to keep the timelines of your project team in mind, if applicable.
```

File: HOW_WE_USE_GITHUB.md [hunk 31]
```diff
@@ -238,17 +238,189 @@ please post details to the [Nucleus forums](https://community.anaconda.cloud/).
...
+
+Task spikes belong to an existing issue. Common use cases for these types of spikes is
+determining if a feature request is viable and if we want to pursue it further, or if a
+reported bug is worth addressing or how best to address it.
+
+When creating a task spike, do the following:
+
+- Under the issues' "tasks" (sometimes called "What should be done?"), add a description
+  of the spike and what needs to be done
+- Add a timebox next to this task (either "low", "medium" or "high")
```
Comment (kathatherine) [inline HOW_WE_USE_GITHUB.md:358]:
```suggestion
- Add a timebox next to this task (either "low", "medium", or "high")
```

File: HOW_WE_USE_GITHUB.md [hunk 32]
```diff
@@ -238,17 +238,189 @@ please post details to the [Nucleus forums](https://community.anaconda.cloud/).
...
+
+---
+
+The example above outlines all the steps necessary in order to create a prototype to present to
+the conda community or project team. For this particular spike, the authors have specified its
+timebox as "high", so as a contributor, you should plan to spend a considerable amount of time
+on this spike.
+
+As someone working on a spike such as this, you are also highly encourage to keep all applicable
+information and discussions on this issue. This helps encourage a culture of asnychronous communication,
```
Comment (kathatherine) [inline HOW_WE_USE_GITHUB.md:417]:
```suggestion
As someone working on a spike like this, you are also highly encourage to keep all applicable
information and discussions within the GitHub issue. This helps encourage a culture of asnychronous communication,
```

**Review by kathatherine — CHANGES_REQUESTED** [review]
I like your rewrites, Travis. Added some suggestions for better clarity and punctuation/grammar fixes.

**Comment by travishathaway** [comment]
@jezdez, will do 👍. Let's make sure we get this done next week though. If it sits any longer in this state, I fear it will never be completed! 😱

**Review by ForgottenProgramme — APPROVED** [review]
